x-pack/plugin/inference/src/test/java/org/elasticsearch/xpack/inference/external/response/ErrorMessageResponseEntityTests.java

```diff
@@ -33,33 +35,52 @@ public void testErrorResponse_ExtractsError() {
         assertThat(error.getErrorMessage(), is("test_error_message"));
     }
 
+    public void testFromResponse_WithOtherFieldsPresent() {
+        String responseJson = """
+            {
+                "error": {
+                    "message": "You didn't provide an API key",
+                    "type": "invalid_request_error",
+                    "param": null,
+                    "code": null
+                }
+            }
+            """;
+
+        ErrorResponse errorMessage = ErrorMessageResponseEntity.fromResponse(
+            new HttpResult(mock(HttpResponse.class), responseJson.getBytes(StandardCharsets.UTF_8))
+        );
+        assertEquals("You didn't provide an API key", errorMessage.getErrorMessage());
+    }
+
     public void testFromResponse_noMessage() {
         String responseJson = """
             {
               "error": {
-                "type": "not_found_error",
+                "type": "not_found_error"
```

Comment:
This was actually causing a parsing failure instead of a missing field failure because of the trailing comma.